scipy/linalg/_decomp_cholesky_update.py

```diff
@@ -0,0 +1,104 @@
+from __future__ import division, print_function, absolute_import
+import numpy
+from numpy import empty_like, sqrt, spacing, tril, triu
+from scipy.linalg import LinAlgError
+
+__all__ = ['cholesky_update']
+
+
+def cholesky_update(R, z, mod_type="+", lower=False, overwrite_R=False,
+               overwrite_z=False, check_finite=True):
+    """
+    Performs a rank-1 update or downdate of Cholesky factors.
+
+    Cholesky rank-1 update and downdate algorithm taken from [1]_ such that
+    an update results in ``D'D = R'R + zz'`` and a downdate results in ``D'D =
+    R'R - zz'``
+
+    Parameters
+    ----------
+    R : (N, N) ndarray
+        The 2D square array from which the triangular part will be used to
+        read the Cholesky factor.The remaining parts are ignored.
+    z : (N,) ndarray
+        A 1D modifier (update or downdate) array
+    mod_type: string, optional
+        The type of modification desired. A "+" indicates an update while a
+        "-" indicates a downdate. Default is "+" (update).
+    lower: bool, optional
+        Whether input array is upper or lower triangular Cholesky
+        factorization. Default is upper-triangular.
+    overwrite_R: bool, optional
+        If set to True the entries of the Cholesky factor array will be
+        modified during the computations instead of creating a new array.
+        Default is False.
+    overwrite_z : bool, optional
+        If set to True the entries of the update array will be modified during
+        the computations instead of creating a new array. Default is False.
+    check_finite : bool, optional
+        Whether to check that the input matrix contains only finite numbers.
+        Disabling may give a performance gain, but may result in problems
+        (crashes, non-termination) if the inputs do contain infinities or NaNs.
+        Default is True.
+
+    Returns
+    -------
+    D : (N, N) ndarray
+        The resulting modified Cholesky factor.
+
+    References
+    ----------
+    .. [1] DOI:10.1007/BF01933218
```

Comment:
Please write the reference in full (name, title, journal) + a `:doi:` link, for which we have Sphinx role so that it will produce a DOI hyperlink. Also, if the paper is gated, is this discussed in e.g. Golub and van Loan?